src/MOI/MOI_wrapper.jl

```diff
@@ -229,7 +231,7 @@ mutable struct Optimizer <: MOI.AbstractOptimizer
 
     # TODO: add functionality to the lower-level API to support querying single
     # elements of the solution.
-    
+    optimizer_called::Bool
```

Comment:
We shouldn't need to add this flag. It's okay for the optimizer to return a previous solution, even if the problem was modified. (JuMP handles these checks so the solvers don't need to: https://jump.dev/JuMP.jl/stable/manual/solutions/#OptimizeNotCalled-errors.)

The test you found was a bit weird, but it wasn't necessarily wrong. Is there a larger problem than the test that you found in your own code?

Reply:
I think I had some errors because I was using Xpress 8.05 and ended up adding unnecessary things.

In fact, the initial problem is testing another Optimizer that allows lazy-cuts to in JuMP interface.
I was testing this package with Xpress.jl applied to SDDP.jl. But Xpress.jl actually needs to call the MOI.get(model, MOI.PrimalStatus()) function each time we request the value of a variable. As this happens a lot in micro-iterations of lazy-cuts, @joaquimg  suggested creating a cache to get the values of all variables, calling MOI.get(model, MOI.PrimalStatus()) only once.

I would just like to create a cache so I don't have to call this function so many times. I removed all the lines about optimizer_called and left only the has_feasible_point cache.

Reply:
Okay, that makes sense. Can you post a screenshot of the tests passing? You don't have permission to run the CI under your account.